src/storage/checkpoint/table_data_writer.cpp

```diff
@@ -137,7 +138,9 @@ void TableDataWriter::FlushSegment(idx_t col_idx) {
 		data_pointer.row_start = last_pointer.row_start + last_pointer.tuple_count;
 	}
 	data_pointer.tuple_count = tuple_count;
-	data_pointers[col_idx].push_back(data_pointer);
+    reinterpret_cast<uint64_t *>(&data_pointer.min_stats)[0] = *stats[col_idx]->minimum.get();
```

Comment:
This assignment seems a bit iffy, I would change this to a memcpy to make it explicit that you are copying the literal bytes.